planetiler-custommap/src/main/java/com/onthegomap/planetiler/custommap/TagValueProducer.java

```diff
@@ -0,0 +1,92 @@
+package com.onthegomap.planetiler.custommap;
+
+import com.onthegomap.planetiler.reader.WithTags;
+import com.onthegomap.planetiler.util.Parse;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.function.UnaryOperator;
+
+public class TagValueProducer {
+
+  static final String STRING_DATATYPE = "string";
+  static final String BOOLEAN_DATATYPE = "boolean";
+  static final String DIRECTION_DATATYPE = "direction";
+  static final String LONG_DATATYPE = "long";
+
+  static final BiFunction<WithTags, String, Object> DEFAULT_GETTER = WithTags::getTag;
+
+  final Map<String, BiFunction<WithTags, String, Object>> valueRetriever = new HashMap<>();
+
+  final Map<String, String> keyType = new HashMap<>();
+
+  static final Map<String, BiFunction<WithTags, String, Object>> dataTypeGetter =
+    Map.of(
+      STRING_DATATYPE, WithTags::getString,
+      BOOLEAN_DATATYPE, WithTags::getBoolean,
+      DIRECTION_DATATYPE, WithTags::getDirection,
+      LONG_DATATYPE, WithTags::getLong
+    );
+
+  static final Map<String, UnaryOperator<Object>> dataTypeParse =
+    Map.of(
+      STRING_DATATYPE, s -> s,
+      BOOLEAN_DATATYPE, Parse::bool,
+      DIRECTION_DATATYPE, Parse::direction,
+      LONG_DATATYPE, Parse::parseLong
+    );
```

Comment:
Can we rename these to tag mapping type or something since it's no longer called data type in the schema?